invenio_drafts_resources/records/models.py

```diff
@@ -17,24 +17,109 @@
 from sqlalchemy_utils.types import UUIDType
 
 
-def ParentRecordMixin(parent_record_cls):
+class ParentRecordMixin:
     """A mixin factory that add the foreign keys to the parent record.
 
-    It is intended to be added to the "child" record class, e.g.:
-    ``class MyRecord(RecordBase, ParentRecordMixin(MyRecordParentClass))``.
+    Usage:
+
+    .. code-block:: python
+
+        class MyRecord(db.Model, RecordMetadataBase, ParentRecordMixin):
+            parent_record_model = ParentRecordMetadata
+    """
+
+    __parent_record_model__ = None
+
+    @declared_attr
+    def parent_id(cls):
+        """Parent identifier."""
+        # We restrict deletion of the parent record in case a record or draft
+        # exists via database-level trigger.
+        return db.Column(UUIDType, db.ForeignKey(
+            cls.__parent_record_model__.id, ondelete="RESTRICT"))
+
+    @declared_attr
+    def parent(cls):
+        """Relationship to parent record."""
+        return db.relationship(cls.__parent_record_model__)
+
+    parent_index = db.Column(db.Integer, nullable=True)
```

Comment:
Comment: after some discussions with Alex/Nico/Jose, then it seemd ``parent_index`` was the best name.

Reply:
What about just ``index``?

Reply:
Some kind of indication that this is for versioning would be nicer or if a one word is much preferred, then `order` or `rank` would be better is my take :)